test/typescript/sample-usage.tsx

```diff
@@ -14,26 +15,44 @@ const testArr: ITodo[] = [
   { text: 'mountain', id: '11' },
   { text: 'air', id: '8' },
   { text: 'plants', id: '9' },
+  { text: 'air', id: '8' },
+  { text: 'air', id: '8' },
 ];
 
-type IProps = {
+type IChildProps = {
   todo: ITodo;
 };
 
-class TestChild extends React.Component<IProps> {
-  shouldComponentUpdate(nextProps: IProps) {
+class TestChild extends React.Component<IChildProps> {
+  shouldComponentUpdate(nextProps: IChildProps) {
     return !equal(this.props, nextProps);
   }
 
   render() {
     const todo = this.props.todo;
+
     return <div>{todo.text}</div>;
   }
 }
 
-class TestContainer extends React.Component {
+type IContainerState = { completed: ITodo[] };
+
+class TestContainer extends React.Component<{}, IContainerState> {
   render() {
-    return testArr.map((item) => <TestChild key={item.id} todo={item} />);
+    const numberOfOverlappingItems: number = useSelector(
+      (state) => this.state.completed.length
```

Comment:
we need to call `useSelector` as the issue reporter does here: https://github.com/FormidableLabs/react-fast-compare/issues/61

The key is the second arg, the `equalityFn`.

Hopefully the test will then _fail_. It should fail in the same way #61 does. Once we've proved that failure, we can modify the types and all the tests should pass.

Reply:
I believe we will need a redux container, as the docs show here: https://react-redux.js.org/api/hooks. The Provider sets the _type_ of the state, which is very important to us :)